js/src/button.js

```diff
@@ -65,56 +62,10 @@ class Button {
   // Public
 
   toggle() {
-    let triggerChangeEvent = true
-    let addAriaPressed = true
-
-    const rootElement = SelectorEngine.closest(
-      this._element,
-      Selector.DATA_TOGGLE
-    )
-
-    if (rootElement) {
-      const input = SelectorEngine.findOne(Selector.INPUT, this._element)
-
-      if (input) {
-        if (input.type === 'radio') {
-          if (input.checked &&
-            this._element.classList.contains(ClassName.ACTIVE)) {
-            triggerChangeEvent = false
-          } else {
-            const activeElement = SelectorEngine.findOne(Selector.ACTIVE, rootElement)
-
-            if (activeElement) {
-              activeElement.classList.remove(ClassName.ACTIVE)
-            }
-          }
-        }
-
-        if (triggerChangeEvent) {
-          if (input.hasAttribute('disabled') ||
-            rootElement.hasAttribute('disabled') ||
-            input.classList.contains('disabled') ||
-            rootElement.classList.contains('disabled')) {
-            return
-          }
-
-          input.checked = !this._element.classList.contains(ClassName.ACTIVE)
-          EventHandler.trigger(input, 'change')
-        }
-
-        input.focus()
-        addAriaPressed = false
-      }
-    }
-
-    if (addAriaPressed) {
-      this._element.setAttribute('aria-pressed',
-        !this._element.classList.contains(ClassName.ACTIVE))
-    }
+    this._element.setAttribute('aria-pressed',
+      !this._element.classList.contains(ClassName.ACTIVE))
 
-    if (triggerChangeEvent) {
-      this._element.classList.toggle(ClassName.ACTIVE)
-    }
+    this._element.classList.toggle(ClassName.ACTIVE)
```

Comment:
So, I guess we don't need the active class any longer, right?

I think we'll need to find another way to add the `aria-pressed` attribute.

Reply:
Yeah, we might be able to nix all the JS—can anyone confirm that? Only reason I could see us leaving some JS for a non-input implementation is quickly toggling active on/off, but I feel like most folks could do that themselves with very little code and their own syntax.